python-api/app/integrations/llm/http.py

```
@@ -0,0 +1,127 @@
+                        payload["stream"] = True
+                    return self.client.chat.completions.create(**payload)
+                if path == "/completions":
+                    payload = json_body.copy() if json_body else {}
+                    if stream:
+                        payload["stream"] = True
+                    return self.client.completions.create(**payload)
+                return self._raw_request(method, path)
+
+            return self._raw_request(method, path)
+        except Exception as e:
+            raise LLMError(self.provider, f"request failed: {e}", raw_response=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)